redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/UIViewFlexContainer.kt

```diff
@@ -91,15 +90,18 @@ internal class UIViewFlexContainer(
 
   private fun alignItems(alignItems: AlignItems) {
     yogaView.rootNode.alignItems = alignItems
-    invalidate()
   }
 
   private fun justifyContent(justifyContent: JustifyContent) {
     yogaView.rootNode.justifyContent = justifyContent
-    invalidate()
   }
 
   private fun invalidate() {
     value.setNeedsLayout()
   }
+
+  override fun onEndChanges() {
+    invalidate()
```

Comment:
Invalidating after all changes are done can save us layout passes and doing intermediate calculations